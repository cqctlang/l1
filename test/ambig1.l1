ns = @names c32le {

	struct S {
		@0x4	struct T t[10];
		@8*10+4;
	};

	struct T {
		@0x4	int x;
		@0x8;
	};


	@0x0 struct S s[10];
};

as = mkzas(1024);

dom = mkdom(ns, as);

@define pr(x)
{
	printf("%d\n", x);
}

dom`s[1].t[2].x = 1;
pr(dom`s[1].t[2].x);
dom`s[1].t[2].x += 1;
pr(dom`s[1].t[2].x);
dom`s[1].t[2].x++;
pr(dom`s[1].t[2].x);
++dom`s[1].t[2].x;
pr(dom`s[1].t[2].x);

pr(&dom`s);
pr(&dom`s[1]);
pr(&dom`s[1].t);
pr(&dom`s[1].t[2]);
pr(&dom`s[1].t[2].x);

p = dom`s;
pr(p[1].t[2].x);
p[1].t[2].x = 1;
pr(p[1].t[2].x);
p[1].t[2].x += 1;
pr(p[1].t[2].x);
p[1].t[2].x++;
pr(p[1].t[2].x);
++p[1].t[2].x;
pr(p[1].t[2].x);

pr(&p[1]);
pr(&p[1].t);
pr(&p[1].t[2]);
pr(&p[1].t[2].x);

@record R { f };

a = [ R(), R() ];
a[1].f = R();
a[1].f.f = [ R(), R(), p ];
a[1].f.f[1].f = R();

a[1].f.f[1].f.f = 1;
pr(a[1].f.f[1].f.f);
a[1].f.f[1].f.f++;
pr(a[1].f.f[1].f.f);
++a[1].f.f[1].f.f;
pr(a[1].f.f[1].f.f);
a[1].f.f[1].f.f += 1;
pr(a[1].f.f[1].f.f);
pr(a[1].f.f[1].f.f++);
pr(a[1].f.f[1].f.f++);
pr(++a[1].f.f[1].f.f);
pr(a[1].f.f[1].f.f);

pr(a[1].f.f[2][1].t[2].x);
pr(&(a[1].f.f[2][1].t[2].x));
