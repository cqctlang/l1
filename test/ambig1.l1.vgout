entry: var params, 26 locals, 0 temps
pr: 1 params, 0 locals, 0 temps
t: var params, 1 locals, 0 temps
x: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
x: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
x: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
x: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
x: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
f: var params, 1 locals, 0 temps
t: var params, 1 locals, 0 temps
x: var params, 1 locals, 0 temps
1
2
3
4
0
84
88
104
108
4
1
2
3
4
84
88
104
108
1
2
3
4
4
5
7
7
4
error: attempt to apply & to non-lvalue
               error	(builtin function)
               entry	(<stdin>:0)
