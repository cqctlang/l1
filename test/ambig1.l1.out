1
2
3
4
0
84
88
104
108
4
1
2
3
4
84
88
104
108
1
2
3
4
4
5
7
7
4
108
